### PR TITLE
Issue 229 - add feature support for Project Lombok. 

### PR DESCRIPTION
**Pull request by wetted**
Issue 229 - add feature support for Project Lombok. 
- Note that lombok must precede micronaut-inject-java for annotation processing.

### CODE REVIEW

**Comment by graemerocher** [comment]
@alvarosanchez is this correct with regards to maven?

File: starter-core/src/main/java/io/micronaut/starter/feature/other/ProjectLombok.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import io.micronaut.starter.options.Language;
+
+import javax.inject.Singleton;
+
+@Singleton
+public class ProjectLombok implements LanguageSpecificFeature {
+
+    private static final String LOMBOK_MAVEN_KEY = "lombok.version";
+    private static final String LOMBOK_GRADLE_KEY = "lombokVersion";
+    private static final String LOMBOK_VERSION = "1.18.16";
```
Comment (graemerocher) [inline starter-core/src/main/java/io/micronaut/starter/feature/other/ProjectLombok.java:32]: Please move this hard coded version to the BOM and call `VersionInfo.getDependencyVersion(String id)` instead

**Review by graemerocher — CHANGES_REQUESTED** [review]

**Comment by alvarosanchez** [comment]
@graemerocher it is targeting 2.1, so for 2.1 is correct.

**Comment by graemerocher** [comment]
We shouldn't be adding new features to 2.1 at this point

**Comment by alvarosanchez** [comment]
Right - @wetted can you please retarget it to 2.2.x?

File: starter-core/src/main/java/io/micronaut/starter/feature/other/ProjectLombok.java (hunk 1 above)
Comment (jameskleeh) [reply starter-core/src/main/java/io/micronaut/starter/feature/other/ProjectLombok.java:32]: @graemerocher This is another thing like testcontainers. I'd like to get group consensus before we add this to the bom

**Review by alvarosanchez — CHANGES_REQUESTED** [review]
Rejecting this until it is rebased and the version is moved to the BOM

**Comment by wetted** [comment]
Closing this PR. I changed it re-target 2.2.x with a new source branch
See https://github.com/micronaut-projects/micronaut-starter/pull/562

**Pull request closed without merging**